kubespawner/objects.py

```diff
@@ -252,6 +241,19 @@ def make_pod(
 
     if lifecycle_hooks:
         lifecycle_hooks = get_k8s_model(V1Lifecycle, lifecycle_hooks)
+
+    security_context = V1SecurityContext()
+    if fs_gid is not None:
+        security_context.fs_group = int(fs_gid)
+    if supplemental_gids is not None and supplemental_gids:
+        security_context.supplemental_groups = [int(gid) for gid in supplemental_gids]
+    if run_as_uid is not None:
+        security_context.run_as_user = int(run_as_uid)
+    if run_as_gid is not None:
+        security_context.run_as_group = int(run_as_gid)
+    if run_privileged:
+        security_context.privileged = True
+
```

Comment:
This will make output become cleaner and tests be less cluttered, without it, a lot of `"securityContext": {}` will show up.

```suggestion
    if [e is None for e in security_context.to_dict().values()]:
        security_context = None
```

Reply:
After this, updating tests to not assume generation of securityContext within the pod specs. But, when either of the following is set, there should be a non-blank entry in the notebook containers securityContext:
fs_gid, supplemental_gids, run_as_uid, run_as_gid, run_privileged

Reply:
Thanks for the suggestion on cleaning the output :+1: .

I've resolved all test failures.

There's still a remaining question around the parameters `fs_gid` (argument: `fs_group`) and `supplemental_gids` (argument: `supplemental_groups`), as those are Pod-only arguments ([V1 Pod Security Context](https://github.com/kubernetes-client/python/blob/v10.0.0/kubernetes/client/models/v1_pod_security_context.py).

We could resurrect the `V1PodSecurityContext` and ensure backward compatibility. Alternatively, deprecate those parameters in favor of _only_ providing users container-level control of the security context.

Reply:
I rebased this PR on current master and added back those as it was easy to not break backward compatibility by doing that.